app/code/Magento/Customer/Controller/Adminhtml/Group.php

```diff
@@ -5,71 +5,48 @@
  */
 namespace Magento\Customer\Controller\Adminhtml;
 
-use Magento\Customer\Api\Data\GroupInterfaceFactory;
-use Magento\Customer\Api\GroupRepositoryInterface;
-
 /**
- * Customer groups controller
+ * Class Group
+ *
+ * @package Magento\Customer\Controller\Adminhtml
  */
 abstract class Group extends \Magento\Backend\App\Action
 {
-    /**
-     * Authorization level of a basic admin session
-     *
-     * @see \Magento\Backend\App\Action\_isAllowed()
-     */
-    const ADMIN_RESOURCE = 'Magento_Customer::group';
 
     /**
-     * Core registry
-     *
      * @var \Magento\Framework\Registry
      */
     protected $_coreRegistry;
-
-    /**
-     * @var GroupRepositoryInterface
-     */
-    protected $groupRepository;
-
-    /**
-     * @var GroupInterfaceFactory
-     */
-    protected $groupDataFactory;
-
-    /**
-     * @var \Magento\Backend\Model\View\Result\ForwardFactory
-     */
-    protected $resultForwardFactory;
-
     /**
-     * @var \Magento\Framework\View\Result\PageFactory
+     * Authorization level of a basic admin session
+     *
+     * @see \Magento\Backend\App\Action\_isAllowed()
      */
-    protected $resultPageFactory;
+    const ADMIN_RESOURCE = 'Magento_Customer::group';
 
     /**
-     * Initialize Group Controller
-     *
      * @param \Magento\Backend\App\Action\Context $context
      * @param \Magento\Framework\Registry $coreRegistry
-     * @param GroupRepositoryInterface $groupRepository
-     * @param GroupInterfaceFactory $groupDataFactory
-     * @param \Magento\Backend\Model\View\Result\ForwardFactory $resultForwardFactory
-     * @param \Magento\Framework\View\Result\PageFactory $resultPageFactory
      */
     public function __construct(
         \Magento\Backend\App\Action\Context $context,
-        \Magento\Framework\Registry $coreRegistry,
-        GroupRepositoryInterface $groupRepository,
-        GroupInterfaceFactory $groupDataFactory,
-        \Magento\Backend\Model\View\Result\ForwardFactory $resultForwardFactory,
-        \Magento\Framework\View\Result\PageFactory $resultPageFactory
+        \Magento\Framework\Registry $coreRegistry
```

Comment:
Considering that \Magento\Framework\Registry is deprecated, @vkublytskyi do you have any suggestions on how to avoid using registry in this case?

Reply:
@vkublytskyi do you have any advice?

Reply:
Actually as this registry dependency was not added in this pull request, it existed in this class before, that refactoring can be postponed